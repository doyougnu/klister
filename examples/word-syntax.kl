#lang "prelude.kl"
(import (shift "prelude.kl" 1))
(import (shift "quasiquote.kl" 1))

(define-macros
  ([gotta-be-word
    (lambda (stx)
      (syntax-case stx
        [(list (_ i))
          (syntax-case i
            [(word int) (pure (word-syntax (+ int int) i))]
            [_ (syntax-error (quasiquote/loc i "bad syntax"))])]
        [_ (syntax-error (quasiquote/loc stx "bad syntax"))]))]))

(example (gotta-be-word 2))
