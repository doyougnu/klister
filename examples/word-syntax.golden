Unknown pattern #[word-syntax.kl:11.14-11.25]<(word int)>
